cmd/mo-service/config.go

```diff
@@ -108,23 +113,64 @@ func (c *Config) createFileService(name string) (fileservice.FileService, error)
 	return nil, fmt.Errorf("file service named %s not set", name)
 }
 
+func (c *Config) createETLFileService(name string) (fileservice.FileService, error) {
+	for _, cfg := range c.FileServices {
+		if strings.EqualFold(cfg.Name, name) {
+			fs, _, err := export.ParseFileService(trace.DefaultContext(), cfg)
+			return fs, err
+		}
+	}
+	return nil, fmt.Errorf("file service named %s not set", name)
+}
+
+func (c *Config) createETLFileServiceAndConfig(name string) (fileservice.FileService, export.FSConfig, error) {
```

Comment:
Not needed. All static file services should be defined in the config file.

Reply:
got it. i agree with fileservice's logic

Reply:
except fileservice, this function also want to get fileservice.Config+S3's access config, which is needed to define external table schema like following:

- there are 2 kind filepath: `{data_dir}/system/error_*.csv` for DISK backend, `{key_prefix}/system/error_*.csv` for S3 backend, both filepath including an item, `{data_dir}` or `{key_prefix}`, come from [[fileservice]] in the config file.

```
# in DISK backend
CREATE EXTERNAL TABLE IF NOT EXISTS error_info(
         statement_id varchar(36),
         span_id varchar(16),
         node_uuid varchar(36) COMMENT "node uuid in MO, which node accept this request",
         node_type varchar(64) COMMENT "node type in MO, enum: DN, CN, LogService;",
         err_code varchar(1024),
         stack varchar(4096),
         timestamp datetime
) infile{"filepath"="{data_dir}/system/error_*.csv","compression"="none"} FIELDS TERMINATED BY ',' ENCLOSED BY '"' LINES TERMINATED BY '\n' IGNORE 0 lines

# in S3 backend
CREATE EXTERNAL TABLE IF NOT EXISTS error_info(
         statement_id varchar(36),
         span_id varchar(16),
         node_uuid varchar(36) COMMENT "node uuid in MO, which node accept this request",
         node_type varchar(64) COMMENT "node type in MO, enum: DN, CN, LogService;",
         err_code varchar(1024),
         stack varchar(4096),
         timestamp datetime
) URL s3option {"endpoint"="s3.us-west-2.amazonaws.com", "access_key_id"="access_key_id", "secret_access_key"="secret_access_key","bucket"="{bucket_name}", "region"="us-west-2", "filepath"="{key_prefix}/system/error_*.csv"} FIELDS TERMINATED BY ',' ENCLOSED BY '"' LINES TERMINATED BY '\n' IGNORE 0 lines
    
```

Reply:
's3option' has been deleted. 
Just use infile {"filepath"="s3:a/b/c"} to access s3 files.

Reply:
got it.